src/main/scala/EnsimePlugin.scala

```diff
@@ -400,9 +401,16 @@ object EnsimePlugin extends AutoPlugin {
       artifact = artifactFilter(classifier = Artifact.DocClassifier)
     ).toSet ++ (ensimeUnmanagedJavadocArchives in config in projectRef).gimme
 
-    // needs to take configurations into account
-    // https://github.com/ensime/ensime-sbt/issues/247
-    val deps = project.dependencies.map(_.project.project).map { n => EnsimeProjectId(n, "compile") }
+    val pattern = "([A-Za-z]+)->([A-Za-z]+)".r
+
+    val deps = project.dependencies.map { d =>
+      val name = d.project.project
+      val config = d.configuration.getOrElse("compile") match {
```

Comment:
btw, `.collect` will turn the `case _` into a `None` and then the fallback will be used.